multiverse-source/src/lib.rs

```diff
@@ -0,0 +1,339 @@
+mod point;
+
+use anyhow::Result;
+use dcspark_blockchain_source::{EventObject, PullFrom, Source};
+use multiverse::{BestBlock, BestBlockSelectionRule, Variant};
+pub use point::Point;
+use std::{
+    fmt::{Debug, Display},
+    hash::Hash,
+};
+
+pub struct Multiverse<K, V, InnerSource> {
```

Comment:
We should narrow the InnerSource type to Source trait as well

Reply:
> let's rename to MultiverseSource, we have the Multiverse already as a storage

yeah, I thought about that too. I was renaming when importing it anyway.

Reply:
> We should narrow the InnerSource type to Source trait as well

I disagree with this, to be honest. I think it's generally a bad practice to put bounds on structs, unless those are required for the code to compile, or you need a field to be an associated type. And I think this is mostly agreed on, since that's the way things work on the std (e.g. `Map` doesn't put an iterator bound on `I`: https://doc.rust-lang.org/std/iter/struct.Map.html).

I could elaborate more on my reasoning about this, but I think I mostly agree with the things said [here](https://stackoverflow.com/a/66369912) (except maybe with the point 1)